TROUBLESHOOTING.md

```diff
@@ -0,0 +1,42 @@
+# Troubleshooting
+
+You may find that the Terraform Provider doesn't always provide the most useful error messages.
+Here are some common errors and their workarounds.
+
+## unexpected state 'CART'
+
+```text
+$ terraform destroy
+... snip ...
+cloudbolt_bp_instance.mycbresource: Destroying... [id=/api/v2/resources/service/35/]
+cloudbolt_bp_instance.mycbresource: Still destroying... [id=/api/v2/resources/service/35/, 10s elapsed]
+
+Error: Error waiting for Job (/api/v2/orders/81/) to complete: unexpected state 'CART', wanted target 'SUCCESS'. last error: %!s(<nil>)
+```
+
+This means that the "Delete" resource action requires approval in CloudBolt.
+
+To resolve this...
+
+1. Go to `https://your.cloudbolt.instance/actions/resource_actions/`
+2. Edit the "Delete" action.
+3. De-select "Requires Aproval".
```

Comment:
```suggestion
3. De-select "Requires Approval".
```

Reply:
It's unclear if this is just a limitation (can't use the cloudbolt approval process?) or if it's just a recommendation - could a user still utilize the approval process? If so, should we link them to the docs for that process?

Reply:
The approval process doesn't really fit in the workflow of Terraform since when you execute Terraform it's going to wait for any provisioned resources to be created. If your Terraform script needs to wait for a person to click "Approve" then it'll be waiting a _very_ long time.

That said we could link to some documentation about custom approval processes. For example, if a Terraform user is making a request in a Terraform Group for some specific Blueprints, auto-approve, otherwise require some number of approvals.